src/com/puppetlabs/mq.clj

```diff
@@ -132,10 +132,17 @@
     (start-broker! (build-embedded-broker brokername dir config))
     (catch java.io.EOFException e
       (log/warn
-        (str "Caught EOFException on broker startup, trying to restart it "
-             "again to see if that solves it. This is probably due to "
-             "KahaDB corruption."))
-      (start-broker! (build-embedded-broker brokername dir config)))))
+       "Caught EOFException on broker startup, trying again."
+       "This is probably due to KahaDB corruption"
+       "(see \"KahaDB Corruption\" in the PuppetDB manual).")
+      (start-broker! (build-embedded-broker brokername dir config)))
+    (catch java.io.IOException e
+      (log/error
+       e "Unable to start broker in" (str (pr-str dir) ".")
```

Comment:
@senior just a simplistic attempt to make paths with whitespace or any other unusual characters a little clearer in some common cases, but it's not rigorous.